docs/guides/serverless-shared-migration-guide.md

```diff
@@ -0,0 +1,266 @@
+
+    Please see the following guide on how to retrieve data from one cluster and store it in another cluster: [Convert a Serverless Instance to a Dedicated Cluster](https://www.mongodb.com/docs/atlas/tutorial/convert-serverless-to-dedicated/)
+
+    Verify that your data is present within the Flex cluster before proceeding.
+4. Delete the Shared-tier cluster by running a destroy command against it.
+    
+    For *mongodbatlas_advanced_cluster*:
+
+    `terraform destroy -target=mongodbatlas_advanced_cluster.this`
+
+    For *mongodbatlas_cluster*:
+
+    `terraform destroy -target=mongodbatlas_cluster.this`
+
+ 5. Remove the resource block for the Shared-tier cluster from your `.tf` file.
+
+### Post-Autoconversion Migration Procedure
+
+Shared-tier clusters will automatically convert in March 2025 or later to Flex clusters in Atlas, retaining all data. 
+
+The following steps resolve the configuration drift in Terraform and does not affect the underlying cluster infrastructure:
+
+1. Find the import IDs of the Flex clusters: `{PROJECT_ID}-{CLUSTER_NAME}`, such as `664619d870c247237f4b86a6-flexClusterName`
+2. Add an import block per cluster to one of your `.tf` files:
+    ```terraform
+    import {
+    to = mongodbatlas_flex_cluster.this
+    id = "664619d870c247237f4b86a6-flexClusterName" # from step 1
+    }
+    ```
+3. Run `terraform plan -generate-config-out=flex_cluster.tf`. This should generate a `flex_cluster.tf` file with your Flex cluster in it.
+4. Run `terraform apply`. You should see the resource(s) imported: `Apply complete! Resources: 1 imported, 0 added, 0 changed, 0 destroyed.`
+5. Remove the "default" fields. Many fields of this resource are optional. Look for fields with a `null` or `0` value.
+6. Re-use existing [Terraform expressions](https://developer.hashicorp.com/terraform/language/expressions). All fields in the generated configuration have static values. Look in your previous configuration for:
+   - variables, for example: `var.project_id`
+   - Terraform keywords, for example: `for_each`, `count`, and `depends_on`
+7. Update the references from your previous cluster resource: `mongodbatlas_advanced_cluster.this.X` or `mongodbatlas_cluster.this.X` to the new `mongodbatlas_flex_cluster.this.X`.
+8. Update any shared-tier data source blocks to refer to `mongodbatlas_flex_cluster`.
+9. Replace your existing clusters with the ones from `flex_cluster.tf` and run 
+
+    `terraform state rm mongodbatlas_advanced_cluster.this`
+
+    or `terraform state rm mongodbatlas_cluster.this`. 
+
+    Without this step, Terraform creates a plan to delete your existing cluster.
+
+10.  Remove the import block created in step 2.
+11.  Re-run `terraform plan` to ensure you have no planned changes: `No changes. Your infrastructure matches the configuration.` 
+
+## From Serverless to Free 
+
+**Please ensure your Serverless instance meets the following requirements to migrate to Free:**
```

Comment:
official doc?

Reply:
To my understanding (from [this conversation with upstream](https://mongodb.slack.com/archives/C07BU4RHNJK/p1731081287931629?thread_ts=1731068246.540079&cid=C07BU4RHNJK)), there won't be any official doc with migration requirement info.

Reply:
Also, there's no explicit mention of constraints (memory, ops sec, etc) in the official docs https://www.mongodb.com/docs/atlas/flex-migration/ it is mentioned “To see which tiers Atlas will migrate your instances to, consult the [All Clusters](https://cloud.mongodb.com/v2#/clusters) page in the Atlas UI.”, and for that the customer will need to access the UI, which some customers can’t do/don’t want to do if they are using TF